pkgs/os-specific/linux/rdma-core/default.nix

```diff
@@ -16,16 +16,20 @@ in stdenv.mkDerivation {
   };
 
   nativeBuildInputs = [ cmake pkgconfig ];
-  buildInputs = [ libnl ethtool libudev python perl ];
+  buildInputs = [ libnl ethtool nettools libudev python perl ];
 
-  postFixup = ''
-    substituteInPlace $out/bin/rxe_cfg --replace ethtool "${ethtool}/bin/ethtool"
+  postPatch = ''
+    substituteInPlace providers/rxe/rxe_cfg.in \
+      --replace '@CMAKE_INSTALL_FULL_SHAREDSTATEDIR@' '/run' \
```

Comment:
how about:

```
cmakeFlags = ["-DCMAKE_INSTALL_RUNDIR=/run"];
```

instead?

Reply:
That does not seem to work. `@CMAKE_INSTALL_FULL_SHAREDSTATEDIR@` still gets substituted by the store path (e.g. /nix/store/vaxhll3n336v6rlh8ql2jqalq1hpykxw-rdma-core-16.1/com/rxe) when I set `CMAKE_INSTALL_RUNDIR=/run`. The cmake doc says that this should be substituted as PREFIX/com 
(https://cmake.org/cmake/help/v3.8/module/GNUInstallDirs.html). Is there another way to override SHAREDSTATEDIR through cmake? I don't see it in the docs.

Reply:
This is strange, when looking at this code: https://github.com/linux-rdma/rdma-core/blob/master/CMakeLists.txt#L93

overriding `CMAKE_INSTALL_RUNDIR` with an absolute path as part of cmake flags should just work.

Reply:
OK I see. I guess the problem is a different one here. The script uses `SHAREDSTATEDIR`, which ends up being PREFIX/com (this seems to be an odd choice to me). If it would use `CMAKE_INSTALL_RUNDIR`, then I guess your suggestions should work.

The reason I need to patch that is the following: rxe_cfg lets one choice if the config should be made persistent or not. Even when I select `-n` (no persistent) it still performs `touch` on the config file (which fails on a location in the nix store). From a nixos perspective the persistent choice should be made through a module. That's why I chose to set this path to /run.

Reply:
Sorry I mixed up variables `-DCMAKE_INSTALL_SHAREDSTATEDIR=/var/lib` should do the trick.
So you would recommend `/run` instead?

Reply:
Unless it would breaks the module, I would keep it as the developer intended it.

Reply:
Also people might want to use it on non-nixos system - does this makes sense?

Reply:
Yes it does make sense. /var/lib is a good choice. Just checked your commit, it works. Thanks!